test/framework/manifest/core_resources.go

```diff
@@ -17,6 +18,11 @@ type ManifestBuilder struct {
 
 	// required when serviceDiscoveryType == CloudMapServiceDiscovery
 	CloudMapNamespace string
+
+	// If set, will not enable ipv6 on any of the pods. This is needed to handle
+	// testing on github actions, where ipv6 is not supported -
+	// https://github.com/actions/runner-images/issues/668.
+	DisableIPv6 bool
```

Comment:
This was a pretty annoying last minute surprise. I'm not happy with how that turned out - there's a lot of ugly logic in here to force ipv6 off in the integration tests to make it run on github actions. Otherwise, any test involving an actual envoy will hang during the proxyinit step displaying any error.

I'm tempted to add a separate flag to the controller itself to disable ipv6 on envoys. I'm worried that any additional work on the tests will result in future developers hitting the same problem, whereas a hard off switch for ipv6 in the controller itself would be more robust.

Reply:
I think both flag or hard off switch are acceptable. Since (I imagine) other customers have hit this ipv6 problem, what do you think about documenting this issue in a readme soemwhere?

Reply:
I think I may introduce a single flag w/ docs in a follow-up patch.